packages/medusa-plugin-auth/src/auth-strategies/auth0/admin.ts

```diff
@@ -0,0 +1,130 @@
+import passport from 'passport';
+import { Strategy as Auth0Strategy } from 'passport-auth0';
+import { ConfigModule, MedusaContainer } from '@medusajs/medusa/dist/types/global';
+import { AUTH_PROVIDER_KEY, ADMIN_AUTH_TOKEN_COOKIE_NAME, TWENTY_FOUR_HOURS_IN_MS } from '../../types';
+import { UserService } from '@medusajs/medusa';
+import { MedusaError } from 'medusa-core-utils';
+import { Router } from 'express';
+import cors from 'cors';
+import { buildCallbackHandler } from '../../core/utils/build-callback-handler';
+import { AUTH0_ADMIN_STRATEGY_NAME, Auth0Options, Profile, ExtraParams } from './types';
+import { PassportStrategy } from '../../core/Strategy';
+
+export class Auth0AdminStrategy extends PassportStrategy(Auth0Strategy, AUTH0_ADMIN_STRATEGY_NAME) {
+	constructor(
+		protected readonly container: MedusaContainer,
+		protected readonly configModule: ConfigModule,
+		protected readonly strategyOptions: Auth0Options
+	) {
+		super({
+			domain: strategyOptions.auth0Domain,
+			clientID: strategyOptions.clientID,
+			clientSecret: strategyOptions.clientSecret,
+			callbackURL: strategyOptions.admin.callbackUrl,
+			passReqToCallback: true,
+			state: true,
+		});
+	}
+
+	async validate(
+		req: Request,
+		accessToken: string,
+		refreshToken: string,
+		extraParams: ExtraParams,
+		profile: Profile
+	): Promise<null | { id: string }> {
+		if (this.strategyOptions.admin.verifyCallback) {
+			return await this.strategyOptions.admin.verifyCallback(
+				this.container,
+				req,
+				accessToken,
+				refreshToken,
+				extraParams,
+				profile
+			);
+		}
+		return await this.defaultValidate(profile);
+	}
+
+	private async defaultValidate(profile: Profile): Promise<{ id: string } | never> {
+		const userService: UserService = this.container.resolve('userService');
+		const email = profile.emails?.[0]?.value;
+
+		if (!email) {
+			throw new MedusaError(
+				MedusaError.Types.NOT_ALLOWED,
+				`Your Auth0 account does not contain a valid email and cannot be used`
+			);
+		}
+
+		const user = await userService.retrieveByEmail(email).catch(() => void 0);
+
+		if (user) {
```

Comment:
I suppose that for the invitation check we are waiting the next refactoring which extract the validate method is that right?